include/leakyvec/leakyvec.hpp

```diff
@@ -142,13 +217,29 @@ class Vec
     }
 
     /// @brief Take ownership of the std::vector back
-    std::vector<T, Alloc> take() noexcept { return std::move(m_inner); }
+    ///
+    /// @note After calling this method, the internal std::vector is left in an empty state.
+    std::vector<T, Alloc> take() noexcept { return std::move(m_inner.inner); }
     /// @brief Get a const reference to the internal std::vector
-    const std::vector<T, Alloc>& as_ref() const noexcept { return m_inner; }
+    const std::vector<T, Alloc>& as_ref() const noexcept { return m_inner.inner; }
     /// @brief Get a mutable reference to the internal std::vector
-    std::vector<T, Alloc>& as_mut() noexcept { return m_inner; }
+    std::vector<T, Alloc>& as_mut() noexcept { return m_inner.inner; }
 
     /// @brief Get the allocator that owns the vector's memory
-    constexpr Alloc get_allocator() const noexcept { return m_inner.get_allocator(); }
+    [[nodiscard]] constexpr Alloc get_allocator() const noexcept { return m_inner.get_allocator(); }
+
+    /// @brief Leak the internal vector as its raw parts
+    ///
+    /// The tuple contains, in order:
+    /// 1. pointer to the start of the vector's data block
+    /// 2. size of the vector (number of initialized elements)
+    /// 3. capacity of the vector (number of allocated elements)
+    /// 4. the allocator that owns the vector's memory block
+    ///
+    /// @note After calling this method, the internal std::vector is left in an empty state.
+    [[nodiscard]] std::tuple<pointer, size_t, size_t, Alloc> leak() noexcept
+    {
+        return m_inner.leak_into_parts();
```

Comment:
Is it actually safe to set the data pointer to `nullptr`? Or do we need to allocate a new empty vector with the allocator to replace it?

(Rust's Vec I think guarantees that the data pointer is non-null)